extension-compose/src/main/java/com/mapbox/maps/extension/compose/MapboxMap.kt

```diff
@@ -0,0 +1,48 @@
+package com.mapbox.maps.extension.compose
+
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.remember
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.platform.LocalContext
+import androidx.compose.ui.platform.LocalInspectionMode
+import androidx.compose.ui.viewinterop.AndroidView
+import com.mapbox.maps.MapInitOptions
+import com.mapbox.maps.MapView
+import com.mapbox.maps.MapboxMap
+
+/**
+ * Entry point for adding a Mapbox Map instance to the Jetpack Compose UI.
+ */
+@Composable
+public fun MapboxMap(
+  /**
+   * Modifier to be applied to the Mapbox map.
+   */
+  modifier: Modifier = Modifier,
+  /**
+   * Defines the initialisation configurations for a [MapboxMap].
+   * It can only be set once and not mutable after the initialisation. Mutating the [MapInitOptions]
+   * during recomposition will result in a [IllegalStateException].
+   */
+  mapInitOptions: MapInitOptions = MapInitOptions(LocalContext.current.applicationContext),
```

Comment:
Wonder if there's some way to express that the `MapInitOptions` is immutable and could not be altered and lead to recomposition (@Immutable annotation? Haven't digged into the semantics but naming is promising)

Reply:
good point, we should take this discovery as a follow up task separately.

Reply:
No, `Immutable` is used for sth else:
> Indicates a type where the value of any properties will never change after the object is constructed, and all methods are referentially transparent

(https://medium.com/androiddevelopers/jetpack-compose-stability-explained-79c10db270c8)

Reply:
Hmm @jush doesn't `a type where the value of any properties will never change after the object is constructed` means properties could not be changed == could not trigger recomposition?

Reply:
I understood that you wanted to apply the `@Immutable` to the `MapInitOptions` class.
If I understand it right, in this case the user could still pass a different `MapInitOptions` which would trigger a new pass for the `MapboxMap`.

Reply:
Ah, probably you are right. Maybe there's some similar way to express this kind of limitations, e.g. limit part of the fields while allowing other to change, imo worth to look into.

Reply:
Another definition can be found in @JorgeCastilloPrz `Jetpack Compose internals` book:
> This annotation is applied over a class as a strict promise for the compiler about all the publicly accessible class properties and fields remaining unchanged after creation. Note that this is a stronger promise than the language val keyword, since val only ensures that the property can’t be reassigned via setter, but it can point to a mutable data structure for example, making our data mutable even if it only has val properties. That would break the expectations of the Compose runtime. In other words, this annotation is needed by Compose essentially because the Kotlin language does not provide a mechanism (a keyword or something) to ensure when some data structure is immutable.

Reply:
TBH, I'm not sure it makes sense to restrict it... In my opinion it shouldn't be much different than if we change the drawable on an [Icon](https://cs.android.com/androidx/platform/frameworks/support/+/androidx-main:wear/compose/compose-material/src/commonMain/kotlin/androidx/wear/compose/material/Icon.kt;l=55?q=composable%20icon) composable call. 

It might be expensive but that's it, we should support that scenario as well as we can.

Reply:
Well I'd say we should very clearly indicate that updating those properties will lead to the whole map recreation (which is slow or very slow for complex styles). 

Another thing is - part of the fields should not mean the whole map reload, e.g. `mapOptions`, `styleUri` or `cameraOptions` should be just quick changes without the MapView recreation.